### PR TITLE
fix: correctly update dynamic binding pathlist 

### LINKED ISSUE #6768: [Bug] Error "TypeError: CustStatusDropdown.selectedOptionValues.reduce is not a function" raised but there is no problem
## Description

I'm getting this error "TypeError: CustStatusDropdown.selectedOptionValues.reduce is not a function" from this logic ANY({{"'{"+CustStatusDropdown.selectedOptionValues.reduce((r, v, i) => r + (i>0?', ':'') + v, '')+"}'"}}) and in the preview panel it says it resolves to "undefined" but it is actually working. I'm able to select one or more values in the dependent multi-select widget and it all works correctly, e.g. the correct results are presented. The only problem is the error being raised and the evaluated window not displaying correctly what is actually being sent to the database. This error is raised if the multi-select widget has values or selected or not. Maybe I shouldn't be using Array reduce and there is a better way to convert dropdown selected values to a comma separated list that doesn't give an error?

CleanShot 2021-08-20 at 10 36 08

CleanShot 2021-08-20 at 10 32 28

### Steps to reproduce the behaviour:

Add steps to reproduce this behaviour, include console / network logs & screenshots

1. Open app
4. See error - see screenshots above

### Important Details

- Version: Cloud
- OS: MacOSX
- Browser: Brave (Chromium)
- Environment: Production

### LINKED ISSUE #7121: [Bug] API Body evaluations doesn't work till body value isn't modified
## Description

[What happened]
[LOOM DEMO](https://www.loom.com/share/6c2bf5c4e6c44baf9e2210805cda727d)

### Steps to reproduce the behaviour:

1. Create 2 APIs (let's name them `A` GET and `B` POST )
2. API `B`'s body params are dependent on A's response value.
3. Make `A` API call and then check `B's` body evaluation.

### Important Details

- Version: [Cloud / Self-Hosted vx.x]
- OS: [e.g.MacOSX]
- Environment [production, release, deploy preview]

### PR DESCRIPTION
**Pull request by ohansFavour**
## Description

Fixes #7121
Fixes #6768

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
Jest

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: fix/dynamic-binding-pathlist-bug 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 54.6 **(0.01)** | 36.45 **(0.01)** | 33.24 **(0.02)** | 55.12 **(0.01)**
 :green_circle: | app/client/src/sagas/ActionSagas.ts | 11.62 **(0.48)** | 0.78 **(0.09)** | 4.17 **(0.47)** | 14.24 **(0.68)**
 :red_circle: | app/client/src/utils/DynamicBindingUtils.ts | 83.8 **(-6.33)** | 69.52 **(-14.02)** | 75.86 **(-3.31)** | 83.02 **(-5.79)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.04 **(-0.22)** | 40.42 **(-0.83)** | 34.48 **(0)** | 55.96 **(-0.26)**</details>

### CODE REVIEW

File: app/client/src/sagas/ActionSagas.test.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
+import { Action, PluginType } from "entities/Action";
```
Comment (hetunandu) [inline app/client/src/sagas/ActionSagas.test.ts:1]: Move test to utils/DynamicBindingsUtils.test.ts

File: app/client/src/utils/DynamicBindingUtils.ts [hunk 2]
```diff
@@ -379,3 +380,36 @@ export const PropertyEvalErrorTypeDebugMessage: Record<
   [PropertyEvaluationErrorType.PARSE]: () => `Could not parse the binding`,
   [PropertyEvaluationErrorType.LINT]: () => `Errors found while evaluating`,
 };
+
+export function getDynamicBindingsChangesSaga(
```
Comment (hetunandu) [inline app/client/src/utils/DynamicBindingUtils.ts:384]: I am unable to figure out what changed in the logic here

File: app/client/src/utils/DynamicBindingUtils.ts [hunk 3]
```diff
@@ -379,3 +380,36 @@ export const PropertyEvalErrorTypeDebugMessage: Record<
...
+  action: Action,
+  value: unknown,
+  field: string,
+) {
+  const bindingField = field.replace("actionConfiguration.", "");
+  let dynamicBindings: DynamicPath[] = action.dynamicBindingPathList || [];
+
+  const valueType = getType(value);
+
+  if (valueType === Types.OBJECT) {
```
Comment (ohansFavour) [inline app/client/src/utils/DynamicBindingUtils.ts:394]: > I am unable to figure out what changed in the logic here (@hetunandu )

The root cause of the bug was `typeof value === "object"` which is true for arrays too.

File: app/client/src/utils/DynamicBindingUtils.ts (hunk 2 above)
Comment (ohansFavour) [reply app/client/src/utils/DynamicBindingUtils.ts:398]: In the latest commit, I have added the case in which 'value' is an array.

**Review by hetunandu — APPROVED** [review]

**Pull request merged**